src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -136,7 +137,7 @@ public IEnumerable<IGraphType> All()
             {
                 lock (_lock)
                 {
-                    var result = _types.FirstOrDefault(x => x.Value.GetType() == type);
+                    var result = _types.SingleOrDefault(x => type.IsAssignableFrom(x.Value.GetType()));
```

Comment:
But then the question arises - what to do if there are several child types? All this seems to be an attempt to correct a design flaw in the wrong place.